src/helpers/cloneIterable.js

```diff
@@ -0,0 +1,34 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Henrique Limas (HenriqueLimas) */
+
+const isArray = require('../core/isArray')
+const isIterable = require('../core/isIterable')
+const isObject = require('../core/isObject')
+
+const err = 'cloneIterable: Iterable required'
+
+function cloneIterable(source) {
+  if (!isIterable(source)) {
+    throw new TypeError(err)
+  }
+
+  if (isArray(source)) {
```

Comment:
For what we're using this for i am unsure that this'll be hit, i think a lot of the methods will check for array first but i think this is still good to have, My only concern is that with this line i think it'll still be holding a reference to the original object right? so if someone was to use this they could be mutating the original values?

Reply:
I needed that condition because I cannot copy an array using the Object.create() and assign, since it will loose the Index. The reference of the internal elements will be the same, but we could instead do a deep clone in some way. Do you think it would be better a deep clone?